src/components/HeaderComponent.tsx

```diff
@@ -1,23 +1,125 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import { withRouter, RouteComponentProps } from 'react-router';
 
 const HeaderComponent = (props: RouteComponentProps) => {
-  const { pathname } = props.location;
-  const path = pathname.split('/')[1];
+  const [path, setPath] = useState('' as string);
+
+  const [subPath, setSubPath] = useState('' as string);
+
+  const [onDropDown, setOnDropDown] = useState(false as boolean);
```

Comment:
Vi kan skrive `useState<boolean>(false)` for å si hvilken type det skal være uten at vi trenger å skrive `as boolean` etc. Tips fra Håkon fra min PR